core/src/main/java/com/epam/esm/service/impl/TagServiceImpl.java

```diff
@@ -28,8 +31,18 @@ public TagDto createTag(TagDto tagDto) {
     }
 
     @Override
-    public List<TagDto> getTags() {
-        return tagRepository.findAll().stream().map(tagConverter::toDTO).collect(Collectors.toList());
+    public Page<TagDto> findPaginated(Pageable pageRequest) {
```

Comment:
Application restrictions
Hibernate specific features.
Spring Data

 public Page<TagDto> findPaginated(Pageable pageRequest) - not allow !